hive-plugins/src/main/java/co/cask/plugin/etl/batch/commons/HiveSchemaConverter.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.plugin.etl.batch.commons;
+
+import co.cask.cdap.api.data.schema.Schema;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import org.apache.hadoop.hive.serde2.objectinspector.PrimitiveObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.hive.hcatalog.common.HCatException;
+import org.apache.hive.hcatalog.data.schema.HCatFieldSchema;
+import org.apache.hive.hcatalog.data.schema.HCatSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * <p>>Hive Schema Converter class to convert {@link Schema} to {@link HCatSchema} and vice versa.</p>
+ * Note: If the {@link HCatSchema} contains non-primitive type then this conversion to {@link Schema} will fail.
+ */
+public class HiveSchemaConverter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HiveSchemaConverter.class);
+
+  /**
+   * Converts a CDAP's {@link Schema} to Hive's {@link HCatSchema} while verifying the fields in the given
+   * {@link Schema} to exists in the table.
+   *
+   * @param schema the {@link Schema}
+   * @return {@link HCatSchema} for the given {@link Schema}
+   */
+  public static HCatSchema toHiveSchema(Schema schema, HCatSchema tableSchema) {
```

Comment:
Hive allows project's while reading/writing data. Hive also support some data types which we don't support in our schema. See the getType for detail. If a user want to read a table which have says void type or date or something which we don't support he cannot read it. Doing this way allow user to give a schema where he drops those fields and read the table. See HiveBatchSource for details.